### PR TITLE
For #12430: Hide the mic when not using Google

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -33,10 +33,10 @@ an hashed version of the Google Advertising ID.
 
 The following metrics are added to the ping:
 
-| Name | Type | Description | Data reviews | Extras | Expiration |
```
Comment (NotWoods) [inline docs/metrics.md:36]: delet this

File: app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt [hunk 2]
```diff
@@ -160,7 +160,11 @@ class SearchFragment : Fragment(), UserInteractionHandler {
             BrowserToolbar.Button(
                 ContextCompat.getDrawable(requireContext(), R.drawable.ic_microphone)!!,
                 requireContext().getString(R.string.voice_search_content_description),
-                visible = { requireContext().settings().shouldShowVoiceSearch && speechIsAvailable() },
+                visible = {
+                    requireContext().settings().shouldShowVoiceSearch &&
+                        speechIsAvailable() &&
+                        currentSearchEngine.searchEngine.name == "Google"
```
Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt:166]: Can we use the `identifier` instead? Does name work on other locales?

Comment (NotWoods) [inline app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt:166]: Would be nice to have this check first so we can skip the settings IO

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt:166]: We have multiple identifiers that refer to Google, so I don't think that works as well as just using the name. I guess I could just check for whether or not the identifier **contains** `google`?

File: docs/metrics.md (hunk 1 above)
Comment (sblatz) [reply docs/metrics.md:36]: I think this is something that needs to be merged. The metrics.md is out of date. I've pulled upstream master and it still has this diff.

File: app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt (hunk 2 above)
Comment (NotWoods) [reply app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt]: contains `google` sounds good, I'd prefer it to the name check

**Pull request merged**